cmd/rook/userfacing/multus/validation/validation.go

```diff
@@ -119,33 +110,51 @@ func init() {
-	runCmd.Flags().StringVar(&validationConfig.PublicNetwork, "public-network", "",
+	runCmd.Flags().StringVar(&validationConfig.PublicNetwork, "public-network", defaultConfig.PublicNetwork,
 		"The name of the Network Attachment Definition (NAD) that will be used for Ceph's public network. "+
 			"This should be a namespaced name in the form <namespace>/<name> if the NAD is defined in a different namespace from the cluster namespace.")
-	runCmd.Flags().StringVar(&validationConfig.ClusterNetwork, "cluster-network", "",
+	runCmd.Flags().StringVar(&validationConfig.ClusterNetwork, "cluster-network", defaultConfig.ClusterNetwork,
 		"The name of the Network Attachment Definition (NAD) that will be used for Ceph's cluster network. "+
 			"This should be a namespaced name in the form <namespace>/<name> if the NAD is defined in a different namespace from the cluster namespace.")
-	runCmd.Flags().IntVar(&validationConfig.DaemonsPerNode, "daemons-per-node", DefaultDaemonsPerNode,
+	runCmd.Flags().IntVar(&validationConfig.DaemonsPerNode, "daemons-per-node", defaultConfig.DaemonsPerNode,
 		"The number of validation test daemons to run per node. "+
 			"It is recommended to set this to the maximum number of Ceph daemons that can run on any node in the worst case of node failure(s). "+
 			"The default value is set to the worst-case value for a Rook Ceph cluster with 3 portable OSDs, 3 portable monitors, "+
 			"and where all optional child resources have been created with 1 daemon such that they all might run on a single node in a failure scenario. "+
 			"If you aren't sure what to choose for this value, add 1 for each additional OSD beyond 3.")
-	runCmd.Flags().StringVar(&validationConfig.NginxImage, "nginx-image", "nginxinc/nginx-unprivileged:stable-alpine",
+	runCmd.Flags().StringVar(&validationConfig.NginxImage, "nginx-image", defaultConfig.NginxImage,
 		"The Nginx image used for the validation server and clients.")
 
+	runCmd.Flags().StringVarP(&validationConfigFile, "config", "c", "",
+		"The validation test config file to use. This cannot be used with other flags.")
+	runCmd.MarkFlagsMutuallyExclusive("config", "timeout-minutes")
+	runCmd.MarkFlagsMutuallyExclusive("config", "namespace")
+	runCmd.MarkFlagsMutuallyExclusive("config", "public-network")
+	runCmd.MarkFlagsMutuallyExclusive("config", "cluster-network")
+	runCmd.MarkFlagsMutuallyExclusive("config", "daemons-per-node")
+	runCmd.MarkFlagsMutuallyExclusive("config", "nginx-image")
+
 	// flags for 'validation cleanup'
 	// none
 }
 
 func runValidation(ctx context.Context) {
-	if validationConfig.PublicNetwork == "" && validationConfig.ClusterNetwork == "" {
-		fmt.Print(`at least one of "--public-network" or "--cluster-network" must be specified`)
-		os.Exit(22 /* EINVAL */)
+	if validationConfigFile != "" {
+		f, err := ioutil.ReadFile(validationConfigFile)
+		if err != nil {
+			fmt.Printf("failed to read config file %q: %s\n", validationConfigFile, err)
+			os.Exit(5 /* EIO */)
```

Comment:
If the config file doesn't exist, error code 2 would be preferred. Worth adding a separate check?

Reply:
There are all kinds of error messages that might be preferred, like EBADF, EACCESS, EISDIR, EMFILE, EROFS, etc. It seems like a fair point that EIO probably isn't a good return value for all the ways this command can fail, so maybe just returning 1 is best. It seems like too much work to translate all the various conditions to linux error numbers.